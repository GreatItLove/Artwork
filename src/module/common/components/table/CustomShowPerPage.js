import React from 'react';
import { DropDownMenu, MenuItem } from 'material-ui';
import PropTypes from 'prop-types';
import _times from 'lodash/times';
import _map from 'lodash/map';

const propTypes = {
  appDispatch: PropTypes.func.isRequired,
  searchFilter: PropTypes.object.isRequired,
};

class CustomShowPerPage extends React.Component {
  onRecordChange = (e, index, value) => {
    const { appDispatch, pagination } = this.props;
    appDispatch(
      this.props.fetchRequest({
        limit: value,
        offset: (pagination.currentPage - 1) * value,
        ...this.props.searchFilter,
      }),
    );
  };
  onPageChange = (e, index, value) => {
    const { appDispatch, pagination } = this.props;
    appDispatch(
      this.props.fetchRequest({
        limit: pagination.recordsPerPage,
        offset: (value - 1) * pagination.recordsPerPage,
        ...this.props.searchFilter,
      }),
    );
  };
  getPaginationInfo = () => {
    const { pagination } = this.props;
    let startRecord = (pagination.currentPage - 1) * pagination.recordsPerPage + 1;
    let endRecord = pagination.recordsPerPage * pagination.currentPage;
    let totalRecord = pagination.totalRecordCount;
    if (pagination.currentPage !== pagination.totalPages && totalRecord)
      return <span>{`${startRecord} - ${endRecord} of ${totalRecord}`}</span>;
    else if (totalRecord) {
      return <span>{`${startRecord} - ${totalRecord} of ${totalRecord}`}</span>
    } else return <span>0 Record</span>;
  };
  render() {
    const { pagination } = this.props;
    return (
      <div className="col-sm-11 show-per-page">
        <div className="row align-items-center">
          <div className="col-md-4 text-left">
            <span className="page">Records per page</span>
            <DropDownMenu
              value={pagination.recordsPerPage}
              onChange={this.onRecordChange}
              underlineStyle={{ width: '30px', margin: '6px 24px' }}
              labelStyle={{ fontSize: '1rem', color: '#979797' }}>
              <MenuItem value="25" primaryText="25" />
              <MenuItem value="50" primaryText="50" />
              <MenuItem value="100" primaryText="100" />
              <MenuItem value="250" primaryText="250" />
            </DropDownMenu>
          </div>
          <div className="col-md-4 text-center">{this.getPaginationInfo()}</div>
          <div className="col-md-4 text-right">
            <span className="page">Page</span>
            <DropDownMenu
              value={pagination.currentPage}
              onChange={this.onPageChange}
              maxHeight={250}
              underlineStyle={{ width: '30px', margin: '6px 24px' }}
              labelStyle={{ fontSize: '1rem', color: '#979797' }}>
              {_map(_times(pagination.totalPages), value => (
                <MenuItem key={value} value={value + 1} primaryText={value + 1} />
              ))}
            </DropDownMenu>
          </div>
        </div>
      </div>
    );
  }
}

CustomShowPerPage.propTypes = propTypes;

export default CustomShowPerPage;
