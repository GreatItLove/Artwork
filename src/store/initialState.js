import { EditorState } from 'draft-js';
import jwt from './jwtStorage';
import selection from './selectionStorage';

let initialAuth = {
  loading: false,
  startPage: null,
  token: false
};

let initialAccount = {
  active: null,
  email: false,
  fullName: false,
  userId: false,
  userType: false,
  loadingData: false,
  loadingError: false,
  isFetchedData: false,
  isTransfer: false
};

let initialCustomer = {
  active: null,
  customerId: null,
  customerType: null,
  email: null,
  logo3: null,
  logoIpad: null,
  organization: null,
  phone: null,
  siteAdministrator: null
};

let initialSettings = {
  bodyName: null,
  curFormat: null,
  dimFormat: null,
  folderName: null,
  ipadUsers: null,
  url: null,
  uses3: null
};
const initialUsers = {
  list: [],
  fetching: false,
  loading: false,
  deleting: false,
  adding: false,
  changingActive: false,
  editing: false,
  resettingPassword: false,
  pagination: {},
  searchFilter: {},
  showingFilter: false,
  fetchingDownloadpath: false,
  downloadPath: '',
  downloading: false,
};

const initialNews = {
  newsData: null,
  newsLoading: true,
  newsError: ''
};

const initialLogs = {
  logsData: [],
  logsLoading: true,
  logsError: ''
};

const initialArtists = {
  list: [],
  pagination: 0,
  selectedCheckBox: [],
  searchFilter: {},
  fetching: false,
  loading: false,
  deleting: false,
  adding: false,
  changingActive: false,
  editing: false,
  resettingPassword: false,
  genres: [],
  categories: [],
  fetchingGenres: false,
  showingFilter: false,
  fetchingArtistInfo: false,
  artistInfo: {},
  editorState_description: EditorState.createEmpty(),
  editorState_biography: EditorState.createEmpty(),
  editorState_resume: EditorState.createEmpty(),
  editorState_statement: EditorState.createEmpty(),
  editorState_pressRelease: EditorState.createEmpty(),
  editorState_articles: EditorState.createEmpty(),
  fetchingDownloadpath: false,
  downloadPath: '',
  downloading: false,
};
const initialContact = {
  list: [],
  category: [],
  adding: false,
  updating: false,
  pagination: 0,
  loading: false,
  fetching: false,
  searchFilter: {},
  showingFilter: false,
  fetchingCategory: false,
  bulkUpdateDialogData: {},
  selectedCheckBox: [],
  specialInterest: [],
  howFound: [],
  customerTypes: [],
  fetchingDownloadpath: false,
  downloadPath: '',
  downloading: false,
};
const initialInvoice = {
  list: [],
  pagination: {
    currentPage: 1,
    recordsPerPage: "25",
    totalPages: 0,
  },
  fetching: false,
  searchFilter: {},
  showingFilter: false,
  selectedCheckBox: [],
  fetchingDownloadpath: false,
  downloadPath: '',
  downloading: false,
};
const isStorageDataExist = jwt.exists();
if (isStorageDataExist) {
  const storageData = jwt.getJWT();
  initialAuth = { ...initialAuth, ...storageData };
}

const isSelectionDataExist = selection.exists();
if (isSelectionDataExist) {
  const storageData = selection.getSelection();
  initialAuth = { ...initialAuth, startPage: storageData };
}
let initialState = {
  auth: initialAuth,
  account: initialAccount,
  customer: initialCustomer,
  settings: initialSettings,
  sysMessage: {
    message: '',
    messageType: ''
  },
  users: initialUsers,
  artists: initialArtists,
  news: initialNews,
  logs: initialLogs,
  contact: initialContact,
  invoice: initialInvoice,
};

export default initialState;
