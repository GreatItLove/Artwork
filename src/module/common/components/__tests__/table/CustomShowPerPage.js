import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CustomShowPerPage from '../../table/CustomShowPerPage';
import initialState from '../../../../../store/initialState';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const store = configureMockStore()({ initialState });
const props = {
  pagination: { currentPage: 1, totalRecordCount: 0, recordsPerPage: "10", },
  appDispatch: jest.fn(),
  searchFilter: {
    limit: 10,
    search: 'test',
  },
};
describe('Snapshot of Custom Show Per Page', () => {
  it('CustomShowPerPage', () => {
    const wrapper = shallow(
      <provider store={store}>
        <CustomShowPerPage {...props} />
      </provider>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
