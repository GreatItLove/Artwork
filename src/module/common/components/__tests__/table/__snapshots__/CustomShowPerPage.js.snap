// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Custom Show Per Page CustomShowPerPage 1`] = `
<provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CustomShowPerPage
    appDispatch={[Function]}
    pagination={
      Object {
        "currentPage": 1,
        "recordsPerPage": "10",
        "totalRecordCount": 0,
      }
    }
    searchFilter={
      Object {
        "limit": 10,
        "search": "test",
      }
    }
  />
</provider>
`;
